nmigen_boards/atlys.py

```diff
@@ -0,0 +1,240 @@
+        Resource("user_btn", 4, Pins("F6",   dir="i"), Attrs(IOSTANDARD="LVCMOS18")), # BTNR
+        Resource("user_btn", 5, Pins("F5",   dir="i"), Attrs(IOSTANDARD="LVCMOS18")), # BTNC
+
+        Resource("user_sw" , 0, Pins("A10",  dir="i"), Attrs(IOSTANDARD="LVCMOS33")), # SW0
+        Resource("user_sw" , 1, Pins("D14",  dir="i"), Attrs(IOSTANDARD="LVCMOS33")), # SW1
+        Resource("user_sw" , 2, Pins("C14",  dir="i"), Attrs(IOSTANDARD="LVCMOS33")), # SW2
+        Resource("user_sw" , 3, Pins("P15",  dir="i"), Attrs(IOSTANDARD="LVCMOS33")), # SW3
+        # SW4 on bank2 done in __init__()
+        # SW5 on bank2 done in __init__()
+        # SW6 on bank2 done in __init__()
+        Resource("user_sw" , 7, Pins("E4",   dir="i"), Attrs(IOSTANDARD="LVCMOS18")), # SW7
+
+        UARTResource(0, rx="A16", tx="B16", attrs=Attrs(IOSTANDARD="LVCMOS33")), # J17/UART
+
+        Resource("ps2", 0, # PS/2 keyboard interface converted from J13 "HOST" USB connector
+            Subsignal("clk",    Pins("P17", dir="i")),
+            Subsignal("dat",    Pins("N15", dir="io")),
+            Attrs(IOSTANDARD="LVCMOS33"),
+        ),
+        Resource("ps2", 1, # PS/2 mouse interface converted from J13 "HOST" USB connector
+            Subsignal("clk",    Pins("N18", dir="i")),
+            Subsignal("dat",    Pins("P18", dir="io")),
+            Attrs(IOSTANDARD="LVCMOS33"),
+        ),
+
+        *SPIFlashResources(0,
+            cs="AE14", clk="AH18", mosi="AF14", miso="AF20", wp="AG21", hold="AG17",
+            attrs=Attrs(IOSTANDARD="LVCMOS25", SLEW="FAST"),
+        ),
+
+        Resource("ddr2", 0,
+            Subsignal("clk",    DiffPairs("G3", "G1", dir="o"),
+                                Attrs(IOSTANDARD="DIFF_SSTL18_II", IN_TERM="NONE")),
+            Subsignal("clk_en", Pins("H7", dir="o")),
+            Subsignal("we",     PinsN("E3", dir="o")),
+            Subsignal("ras",    PinsN("L5", dir="o")),
+            Subsignal("cas",    PinsN("K5", dir="o")),
+            Subsignal("a",      Pins("J7 J6 H5 L7 F3 H4 H3 H6 D2 D1 F4 D3 G6", dir="o")),
+            Subsignal("ba",     Pins("F2 F1 E1", dir="o")),
+            Subsignal("dqs",    DiffPairs("P2 L4", "P1 L3", dir="o"),
+                                Attrs(IOSTANDARD="DIFF_SSTL18_II")),
+            Subsignal("dq",     Pins("L2 L1 K2 K1 H2 H1 J3 J1 M3 M1 N2 N1 T2 T1 U2 U1", dir="io")),
+            Subsignal("dm",     Pins("K4 K3", dir="o")),
+            Subsignal("odt",    Pins("K6", dir="o")),
+            Attrs(IOSTANDARD="SSTL18_II", SLEW="FAST"),
+        ),
+
+        Resource("eth_gmii", 0,
+            Subsignal("rst",     PinsN("G13", dir="o")),
+            Subsignal("int",     PinsN("L16", dir="o")),
+            Subsignal("mdio",    Pins("N17", dir="io")),
+            Subsignal("mdc",     Pins("F16", dir="o")), # Max 8.3MHz
+            Subsignal("gtx_clk", Pins("L12", dir="o")),
```

Comment:
What's `gtx_clk`?

Reply:
It standard signal defined for [GMII](https://en.wikipedia.org/wiki/Media-independent_interface#Gigabit_media-independent_interface). In Gbit mode this is a 125MHz clock provided from MAC to PHY for tx. In 10/100Mbit the 25/2.5MHz clock tx_clk from PHY to MAC is used as for MII.